packages/x-components/tests/e2e/cucumber/keyboard-navigation.spec.ts

```diff
@@ -11,7 +11,9 @@ type Direction = 'above' | 'below' | 'on the left' | 'on the right';
 
 // Scenario 1
 And('{string} element position is stored', (focusableElement: string) => {
-  cy.getByDataTest(focusableElement).focus().as('originalElement');
+  return focusableElement === 'search-input'
+    ? cy.getByDataTest(focusableElement).focus().as('originalElement')
+    : cy.getByDataTest(focusableElement).last().focus().as('originalElement');
 });
 
 Then('next element position is "{direction}"', (expectedPosition: Direction) => {
```

Comment:
Something is wrong with the `XXX out of bounds is reached` as if you remove the 

```javascript
.trigger('keydown', { key: 'ArrowDown', force: true })
.trigger('keydown', { key: 'ArrowRight', force: true })
```

They tests pass exactly the same, please take a look.

Reply:
Mmmm don't know if I understood correctly, but that's the point. What I am trying to check here is that once you have the focus on the first / last possible navigable position, pressing the pertinent arrow keys will be the same as not pressing anything at all since the bound have been reached. Thus, if you are on the first position (`originalElement`) and you use left and/or up keys to move to a _previous_ one (`targetElement`), it won't be possible and `originalElement` and `targetElement` will have the same value, so the equivalent is as if you pressed nothing at all.